src/ymlmd/yml2md.py

```diff
@@ -0,0 +1,162 @@
+"""
+
+# Standard Python Libraries
+import logging
+import sys
+from typing import Any
+
+# Third-Party Libraries
+import dateparser
+import docopt
+from schema import And, Schema, SchemaError, Use
+import yaml
+
+# cisagov Libraries
+from mdyml import DEFAULT_CVE_ID
+
+from . import __version__
+
+Software = list[dict[str, Any]]
+
+
+def load(filename: str) -> Software:
+    """Load the YML into a Python dictionary."""
+    ans = None
+    with open(filename, "r") as f:
+        ans = yaml.safe_load(f)
+
+    return ans
+
+
+def calculate_status(software: Software) -> Software:
+    """Calculate status field for each software entry."""
+    for s in software:
+        # We are only worrying about the main (default) CVE for now.
+        default_cve = s["cves"][DEFAULT_CVE_ID]
+        if len(default_cve["affected_versions"]) == 0:
+            if len(default_cve["fixed_versions"]) == 0:
+                if default_cve["investigated"]:
+                    s["status"] = "Not Affected"
+                else:
+                    s["status"] = "Unknown"
+            else:  # fixed_versions is not empty
+                s["status"] = "Fixed"
+        else:  # affected versions is not empty
+            if len(default_cve["fixed_versions"]) == 0:
+                s["status"] = "Affected"
+            else:
+                s["status"] = "Fixed"
+    return software
+
+
+def generate_markdown(software: Software) -> None:
+    """Generate Markdown manually."""
+    header_fields = [
+        "Vendor",
+        "Product",
+        "Affected Versions",
+        "Patched Versions",
+        "Status",
+        "Vendor Links",
+        "Notes",
+        "References",
+        "Reporter",
+        "Last Updated",
+    ]
+
+    # Print header and delimiter rows
+    for row_contents in [header_fields, ["-" * len(field) for field in header_fields]]:
+        print(f'| {" | ".join(row_contents)} |')
+
+    # Print table converting lists to comma separated strings
+    for i, s in enumerate(software, start=1):
+        default_cve = s["cves"][DEFAULT_CVE_ID]
+        try:
+            print(
+                "| {vendor} | {product} | {affected_versions} | {patched_versions} | {status} | {vendor_links} | {notes} | {references} | {reporter} | {last_updated} |".format(
+                    vendor=s["vendor"],
+                    product=s["product"],
+                    affected_versions=",".join(
+                        [x for x in default_cve["affected_versions"] if len(x) != 0]
+                    ),
+                    patched_versions=",".join(
+                        [x for x in default_cve["fixed_versions"] if len(x) != 0]
+                    ),
+                    status=s["status"],
+                    # convert vendor links to Markdown links if they start
+                    # with http
+                    vendor_links=", ".join(
+                        [
+                            f"[link]({link})" if link.startswith("http") else link
```

Comment:
I would kind of like to massage URLs to HTTPS, but what does everyone think?

Reply:
So far, we are not really modifying user-entered data for our MVP.  Please make an issue about this and maybe someone can get it done later.